### PR TITLE
move to go modules

### PR DESCRIPTION
**Pull request by qinqon**
Signed-off-by: Quique Llorente <ellorent@redhat.com>

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -107,10 +114,32 @@ func main() {
 		os.Exit(1)
 	}
 
-	// Create Service object to expose the metrics port.
-	_, err = metrics.ExposeMetricsPort(ctx, metricsPort)
+	if err = serveCRMetrics(cfg); err != nil {
```
Comment (phoracek) [inline cmd/manager/main.go]: Why do we need these in this PR?

File: hack/install-github-release.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash -xe
```
Comment (phoracek) [inline hack/install-github-release.sh]: How is this change related to the rest? Would you please split the PR into multiple commits (or multiple PRs, but won't fight for it). Moving to newer operator-sdk, go.mod, this little script.

File: cmd/manager/main.go (hunk 1 above)
Comment (qinqon) [reply cmd/manager/main.go]: operator-sdk 0.12.0 has this at generated code, I have updated it to make it compile.

File: hack/install-github-release.sh (hunk 2 above)
Comment (qinqon) [reply hack/install-github-release.sh]: We cannot, proper support for operator-sdk is done at 0.12.0 so it have to be at the same PR, although I have to remove this script.

File: automation/check-patch.e2e-k8s.sh [hunk 3]
```diff
@@ -15,7 +15,6 @@ main() {
     export KUBEVIRT_PROVIDER='k8s-1.14.6'
 
     source automation/check-patch.e2e.setup.sh
-    cd ${TMP_PROJECT_PATH}
```
Comment (phoracek) [inline automation/check-patch.e2e-k8s.sh]: Why do you remove this? It is a part of the kubevirtci scripting. It makes sure that you can execute the automation/ test script on any environment and it will create a temporary project directory with everything working, even if you did not have a working GOPATH before.

File: automation/check-patch.e2e.setup.sh [hunk 4]
```diff
@@ -1,27 +1,9 @@
-# Prepare environment for kubernetes-nmstate end to end testing. This includes temporary Go paths and binaries.
+# Prepare environment for kubernetes-nmstate end to end testing.
 #
-# source automation/check-patch.e2e.setup.sh
-# cd ${TMP_PROJECT_PATH}
 
-echo 'Setup Go paths'
-export GOROOT=/tmp/kubernetes-nmstate/go/root
```
Comment (phoracek) [inline automation/check-patch.e2e.setup.sh]: With this the script would kill my Go environment if executed locally. On top of that, it would not work in an environment where Go is not set up yet. The point of these scripts was to allow it to run anywhere without dependencies.

File: automation/check-patch.e2e-k8s.sh (hunk 3 above)
Comment (qinqon) [reply automation/check-patch.e2e-k8s.sh]: Adding them it back.

File: automation/check-patch.e2e.setup.sh (hunk 4 above)
Comment (qinqon) [reply automation/check-patch.e2e.setup.sh]: going back to them

File: hack/install-github-release.sh (hunk 2 above)
Comment (phoracek) [reply hack/install-github-release.sh]: You promised to remove this :(

File: Makefile [hunk 5]
```diff
@@ -78,7 +81,7 @@ test/unit: $(GINKGO)
 	INTERFACES_FILTER="" NODE_NAME=node01 $(GINKGO) $(unit_test_args) ./pkg/
 
 test/e2e: $(OPERATOR_SDK)
-	$(OPERATOR_SDK) test local ./test/e2e \
+	unset GOFLAGS && $(OPERATOR_SDK) test local ./test/e2e \
```
Comment (phoracek) [inline Makefile]: What is the unset good for?

Comment (qinqon) [reply Makefile]: operator-sdk is adding another mod=vendor so it fails.

Comment (phoracek) [reply Makefile]: Please explain it in an inline comment for our future selves.

**Comment by phoracek** [comment]
ci test please

File: cmd/manager/main.go (hunk 1 above)
Comment (phoracek) [inline cmd/manager/main.go]: Feels bad to have metrics that we never even tried. But it is not a regression compared to the original code, just complaining.

File: hack/install-github-release.sh (hunk 2 above)
Comment (qinqon) [reply hack/install-github-release.sh]: Broken promises, let's remove it :-)

File: cmd/manager/main.go (hunk 1 above)
Comment (qinqon) [reply cmd/manager/main.go]: Let's try to simplify not used stuff at follow-up PRs
